Tests/DatadogIntegrationTests/UITestsHelpers.swift

```diff
@@ -8,23 +8,16 @@ import XCTest
 
 /// Convenient interface to navigate through Example app's main screen.
 class ExampleApplication: XCUIApplication {
-    /// Launches the app by mocking feature endpoints.
+    /// Launches the app by providing mock server configuration.
     func launchWith(
         testScenario: TestScenario.Type,
-        logsEndpointURL: URL? = nil,
-        tracesEndpointURL: URL? = nil,
-        rumEndpointURL: URL? = nil,
-        customEndpointURL: URL? = nil
+        serverConfiguration: HTTPServerMockConfiguration
     ) {
         launchArguments = [Environment.Argument.isRunningUITests]
 
         var variables: [String: String] = [:]
         variables[Environment.Variable.testScenarioIdentifier] = testScenario.envIdentifier()
-
-        logsEndpointURL.flatMap { variables[Environment.Variable.logsEndpoint] = $0.absoluteString }
-        tracesEndpointURL.flatMap { variables[Environment.Variable.tracesEndpoint] = $0.absoluteString }
-        rumEndpointURL.flatMap { variables[Environment.Variable.rumEndpoint] = $0.absoluteString }
-        customEndpointURL.flatMap { variables[Environment.Variable.customEndpoint] = $0.absoluteString }
+        variables[Environment.Variable.serverMockConfiguration] = serverConfiguration.toEnvironmentValue
```

Comment:
i didn't understand this part: first we serialize `serverConf` into base64 and then deserialize it into a Swift object? why do we do that?

Reply:
The `ProcessInfo.environment` expects `[String : String]`, so we need to represent the configuration as JSON `String`. I used Base64 additionally to limit the characters to ASCII - depending on how Xcode passes those values to the Simulator process, I was afraid that some characters in JSON may break the input (but I didn't test if it breaks without Base64).